### PR TITLE
Final project PR review

### PR DESCRIPTION
**Pull request by francescoaracu**
## Changelog

* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
This PR introduces all the new features requested in the issues; all the features are implemented via multiple commits which are detailed (also with photos) in the related issues.

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
No, this PR is the final PR that contains all the new and modified files that are needed to run the web server without any issues.

* **Other information**: (does the pr fix some issues? Tag them with #<issue id>)
List of implemented features:
- Image histogram feature (#1)
- Image transformation feature (#2)
- Download results (in JSON and PNG) buttons feature (#3)
- Upload image feature (#4)

@maurapintor

### CODE REVIEW

File: Readme.md
Comment (maurapintor) [inline Readme.md:9]: If I accept this change, it will change the readme of my project. I don't think this is a good idea. Could you remove it?

File: Readme.md
Comment (maurapintor) [inline Readme.md:41]: What is this script? It was not requested, and it is not listed in the changes. Why do we need it?

File: app/utils/classifications_uploaded_utils.py
Comment (maurapintor) [inline app/utils/classifications_uploaded_utils.py:13]: Why this check? is it necessary?

File: prepare_models.py
Comment (maurapintor) [inline prepare_models.py:8]: I don't get all these `@staticmethod`s that have been added. I am not requesting to revert them, but it's generally not a good practice to apply changes like these in a "new feature" pull request.

**Review by maurapintor — CHANGES_REQUESTED** [review]
The code is easy to read and well documented. I left a few comments to fix before merging. Please, push your changes in the same branch for which you opened the pull request, without closing the PR. In this way, I will keep track of the files I already viewed and I won't have to start the review from scratch.

File: Readme.md
Comment (lmolinario) [reply Readme.md:41]: I added the script to start runserver.py and the workers (worker.py and worker_histo.py) faster.
the idea was to start connection, queue and redis tests as well. No one actually requested it so it will be removed.

File: prepare_models.py
Comment (lmolinario) [reply prepare_models.py:8]: I put them in the beginning, when trying to figure out how the code worked. They will all be removed

File: app/utils/classifications_uploaded_utils.py
Comment (IlPodd) [reply app/utils/classifications_uploaded_utils.py:13]: Since the "secure_filename" function provided by the" werkzeug.utils" package does not sanitize filenames with multiple extensions, such as 'image.php.png', it is necessary to implement additional checks beyond the "secure_filename" function in order to prevent malicious uploads on the web server. Although this check was not required, it was implemented to prioritize web security over functionality.
 
### Below is shown the output of the above mentioned string given by the "secure_filename" function:

immagine_2023-06-19_151807380

**Comment by francescoaracu** [comment]
@maurapintor we have committed all the things that you asked us. Let us know if it's all good now.

File: tests/test_redis.py
Comment (maurapintor) [inline tests/test_redis.py:1]: It's not a good practice to put subprocess popen in a unit test. It's better to leave this file as it was before the changes.

**Review by maurapintor — CHANGES_REQUESTED** [review]
I left just a minor change before closing the PR. Please, fix that and we can merge.

File: tests/test_redis.py
Comment (lmolinario) [reply tests/test_redis.py:1]: Sure!I'll do it immediately!

**Comment by francescoaracu** [comment]
Seems to be all good now! @maurapintor

**Review by maurapintor — APPROVED** [review]
Looks good now. I will merge it into the main repo. Good job! 👍

**Comment by maurapintor** [comment]
One final comment. Could you update the text of the PR? You should remove all references to `start.py` and also add the list of features you implemented. You can also reference the issues with the GitHub automatic links (just type # and the number of the issue, e.g., #1).

**Comment by francescoaracu** [comment]
Of course! Doing it right now @maurapintor

**Comment by maurapintor** [comment]
Looks good now. Merging the PR to the main branch of the forked project.

**Pull request merged**
